File: Algorithms/169 - Majority Element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        #initialize
        count = list()
        count.append(1)
         
        nums.sort()
         
        i = 1
        while i < len(nums):
            if nums[i] == nums[i - 1]:
                count.append(count[i - 1] + 1)
            else:
                count.append(1)
            #break condition
            if count[i] > len(nums) / 2:
                return nums[i]
            i += 1
        return nums[0]

File: Algorithms/cpp/169 - Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        // pre
        sort(nums.begin(), nums.end());
        
        // var
        int count = 1;
        for (int i = 1; i < nums.size(); i++) {
            count = nums[i - 1] == nums[i] ? count + 1 : 1;
            if (count > nums.size() / 2)
                return nums[i];
        }
        return nums[0];
    }
};